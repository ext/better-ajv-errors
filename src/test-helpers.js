import { readFile } from "node:fs/promises";
import path from "node:path";

export async function readJsonFile(filePath) {
	const content = await readFile(filePath, "utf8");
	return JSON.parse(content);
}

export async function getSchemaAndData(name, fixturesPath) {
	const schemaPath = path.join(fixturesPath, name, "schema.json");
	const schema = await readJsonFile(schemaPath);
	const dataPath = path.join(fixturesPath, name, "data.json");
	const json = await readFile(dataPath, "utf8");
	const data = JSON.parse(json);
	return [schema, data, json];
}
