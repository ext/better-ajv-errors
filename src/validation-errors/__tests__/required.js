import path from "node:path";
import { parse } from "@humanwhocodes/momoa";
import { getSchemaAndData } from "../../test-helpers";
import RequiredValidationError from "../required";

const fixturePath = path.join(__dirname, "..", "__fixtures__");

describe("Required", () => {
	it("prints correctly for missing required prop", async () => {
		const [schema, data] = await getSchemaAndData("required", fixturePath);
		const jsonRaw = JSON.stringify(data, null, 2);
		const jsonAst = parse(jsonRaw);

		const error = new RequiredValidationError(
			{
				keyword: "required",
				dataPath: "/nested",
				schemaPath: "#/required",
				params: { missingProperty: "id" },
				message: `should have required property 'id'`,
			},
			{ data, schema, jsonRaw, jsonAst },
		);

		expect(error.print()).toMatchSnapshot();
	});
});
