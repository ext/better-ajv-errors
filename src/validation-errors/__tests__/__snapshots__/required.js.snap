// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Required prints correctly for missing required prop 1`] = `
[
  "<red><bold>REQUIRED</intensity> should have required property 'id'</color>
",
  "  1 | {
> 2 |   "nested": {}
    |             ^ <magenta>id</color> is missing here!
  3 | }",
]
`;
